frontend/src/app/event/event-page/event-page.component.ts

```diff
@@ -7,37 +7,67 @@
  * @license MIT
  */
 
-import { Component, HostListener, OnInit } from '@angular/core';
+import {
+  Component,
+  HostListener,
+  OnInit,
+  inject,
+  OnDestroy
+} from '@angular/core';
 import { profileResolver } from 'src/app/profile/profile.resolver';
-import { eventResolver } from '../event.resolver';
-import { ActivatedRoute } from '@angular/router';
+import { ActivatedRoute, ActivationEnd, Params, Router } from '@angular/router';
 import { Profile } from 'src/app/profile/profile.service';
 import { Event } from '../event.model';
 import { DatePipe } from '@angular/common';
-import { EventFilterPipe } from '../event-filter/event-filter.pipe';
 import { EventService } from '../event.service';
 
+import { PaginatedEvent } from 'src/app/pagination';
+import {
+  Subject,
+  Subscription,
+  debounceTime,
+  distinctUntilChanged,
+  filter,
+  tap
+} from 'rxjs';
+
 @Component({
   selector: 'app-event-page',
   templateUrl: './event-page.component.html',
   styleUrls: ['./event-page.component.css']
 })
-export class EventPageComponent implements OnInit {
+export class EventPageComponent implements OnInit, OnDestroy {
+  public page: PaginatedEvent<Event>;
+  public startDate = new Date();
+  public endDate = new Date(new Date().setMonth(new Date().getMonth() + 1));
+  public today: boolean = true;
+
+  private static EventPaginationParams = {
+    order_by: 'time',
+    ascending: 'true',
+    filter: '',
+    range_start: new Date().toLocaleString('en-GB'),
+    range_end: new Date(
+      new Date().setMonth(new Date().getMonth() + 1)
+    ).toLocaleString('en-GB')
+  };
+
   /** Route information to be used in App Routing Module */
   public static Route = {
     path: '',
     title: 'Events',
     component: EventPageComponent,
     canActivate: [],
-    resolve: { profile: profileResolver, events: eventResolver }
+    resolve: {
+      profile: profileResolver,
+      page: () =>
+        inject(EventService).list(EventPageComponent.EventPaginationParams)
+    }
```

Comment:
Since this code is a bit complex, it might be good to include comments / light documentation!